gammapy/catalog/core.py

```diff
@@ -257,18 +269,50 @@ def _make_source_object(self, index):
             name_extended = data["Source_Name"].strip()
         else:
             name_extended = None
+
+        name_spectral = self._get_name_spectral(data)
+
         try:
             idx = self._lookup_extended_source_idx[name_extended]
             data_extended = table_row_to_dict(self.extended_sources_table[idx])
         except (KeyError, AttributeError):
             data_extended = None
 
-        source = self.source_object_class(data, data_extended)
+        try:
+            idx = self._lookup_spectral_source_idx[name_spectral]
+            data_spectral = table_row_to_dict(self.spectral_table[idx])
+        except (KeyError, AttributeError):
+            data_spectral = None
+
+        source = self.source_object_class(data, data_extended, data_spectral)
         return source
 
+    def _get_name_spectral(self, data):
+        if "Source_name" in data:
+            name_spectral = data["Source_name"].strip()
+        else:
+            name_spectral = None
+        return name_spectral
+
+    def _get_spectral_table_source_name_key(self):
+        return self._source_name_key
```

Comment:
This does not seem necessary. The result is longer than the direct call.
If this is there for daughter classes, maybe it could be a simple attribute rather than a function?

Reply:
I honestly cannot remember why I did that.